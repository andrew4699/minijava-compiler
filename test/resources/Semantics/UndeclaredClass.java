class Factorial{
        public static void main(String[] a){
                System.out.println(5);
        }
}

class A {
        public int x(){
                C c;
    c = new C();
    return 5;
        }
}
