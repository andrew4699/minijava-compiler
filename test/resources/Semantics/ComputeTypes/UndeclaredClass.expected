[ComputeTypesVisitor - 10]: Cannot create object of unknown class C
Program
  MainClass Factorial{unknown} (line 1)
    Print (line 3)
      5{int}
  Class A{class} (line 7)
    MethodDecl x{unknown} (line 8)
      returns int
      VarDecl C c{unknown} (line 9)
      c{unknown} = new C{unknown}(){unknown}
      Return 5{int} (line 11)
